bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java

```diff
@@ -603,7 +603,9 @@ public static void setDeviceZoom (int nativeDeviceZoom) {
 
 	DPIUtil.deviceZoom = deviceZoom;
 	System.setProperty("org.eclipse.swt.internal.deviceZoom", Integer.toString(deviceZoom));
-	if (deviceZoom != 100 && autoScaleMethodSetting == AutoScaleMethod.AUTO) {
+	boolean isGtk = SWT.getPlatform().equals("gtk");
+	if (!(isGtk && deviceZoom == 100) // in GTK, preserve the current mode when switching to a 100% monitor
+			&& autoScaleMethodSetting == AutoScaleMethod.AUTO) {
```

Comment:
The functionality seems to be correct. I would only propose to clean up the code a bit to not have such a complex if statement with an additional comment in there. What about something more simple like:
```suggestion
	boolean preserveScalingMethod = SWT.getPlatform().equals("gtk") && deviceZoom == 100;
	if (!preserveScaling && autoScaleMethodSetting == AutoScaleMethod.AUTO) {
```

Reply:
Done in 8dace1dcbfe1ad4dec23bd3eee6a40fdaeac7645